server/controllers/finance/journal/index.js

```diff
@@ -16,12 +16,61 @@ const db = require('../../../lib/db');
 const core = require('./core');
 const uuid = require('node-uuid');
 const journal = require('../journal/voucher');
+const NotFound   = require('../../../lib/errors/NotFound');
+const _ = require('lodash');
 
 // expose to the api
 exports.list = list;
-
+exports.getTransaction = getTransaction;
 exports.reverse = reverse;
 
+/**
+ * Looks up a transaction by trans_id.
+ *
+ * @param {String} trans_id - the transaction ID
+ * @returns {Promise} object - a promise resolvinng to the part of transaction object.
+ */
+function lookupTransaction(trans_id) {
```

Comment:
This should use the record_uuid.  Binary comparison is much faster than string comparison.